### PR TITLE
port toranj test-013-off-mesh-route-traffic.py

### PR DESCRIPTION
**Pull request by priyankaCh22**

### CODE REVIEW

File: silk/tests/openthread/ot_test_off_mesh_route_traffic.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            message = random_string(10)
+            src_address = ""
+            if src_type == AddressType.Prefix:
+                address = src.find_ip6_address_with_prefix(ON_MESH_PREFIX)
+                self.wait_for_completion(self.device_list)
+                src_address = f"{address}%{src.netns}"
+
+            dst.receive_udp_data(port, message, timeout)
+            time.sleep(delay)
+            src.send_udp_data(dst_address, port, message, src_address)
```
Comment (priyankaCh22) [inline silk/tests/openthread/ot_test_off_mesh_route_traffic.py:228]: #68

File: silk/tests/openthread/ot_test_off_mesh_route_traffic.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    def test03_verify_off_mesh_routes(self):
+        # Add on-mesh prefix
+        self.r1.config_gateway(ON_MESH_PREFIX)
+
+        # The off-mesh-routes are added as follows:
+        # - `r1` adds OFF_MESH_ROUTE_1,
+        # - `r2` adds OFF_MESH_ROUTE_2,
+        # - `fed1` adds OFF_MESH_ROUTE_3.
+
+        self.r1.add_route_using_prefix(OFF_MESH_ROUTE_1)
```
Comment (tzm41) [inline silk/tests/openthread/ot_test_off_mesh_route_traffic.py:167]: Getting: `AttributeError: 'ThreadDevBoard' object has no attribute 'add_route_using_prefix'`.

File: silk/tests/openthread/ot_test_off_mesh_route_traffic.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        self.r2.join(self.network_data, "router")
+        self.wait_for_completion(self.device_list)
+
+        self.fed1.join(self.network_data, "end-node")
+        self.wait_for_completion(self.device_list)
+
+        self.sed2.join(self.network_data, "sleepy-end-device")
+        self.sed2.set_sleep_poll_interval(POLL_INTERVAL)
+        self.wait_for_completion(self.device_list)
+
```
Comment (qilinkang) [inline silk/tests/openthread/ot_test_off_mesh_route_traffic.py:154]: Pls add code here to check if r2 becomes a "router"

File: silk/tests/openthread/ot_test_off_mesh_route_traffic.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+        timeout = 5
+        delay = 1
+        for i, (src, dst, src_type, dst_address) in enumerate(addresses):
+            port = random.randint(10000 + i * 100, 10099 + i * 100)
+            message = random_string(10)
+            src_address = ""
+            if src_type == AddressType.Prefix:
+                address = src.find_ip6_address_with_prefix(ON_MESH_PREFIX)
+                self.wait_for_completion(self.device_list)
+                src_address = f"{address}%{src.netns}"
```
Comment (qilinkang) [inline silk/tests/openthread/ot_test_off_mesh_route_traffic.py]: I think we only need to attach src.netns for LLA address.

File: silk/tools/wpan_table_parser.py [hunk 5]
```diff
@@ -446,7 +446,8 @@ def parse_list(list_string):
     # For each line, skip the first two characters (which are `\t"`) and last character (`"`), then split the string
     # using whitespace as separator. The first entry is the IPv6 address.
     #
-    return [line[2:-1].split()[0] for line in list_string.split("\n")[1:-1]]
+    # return [line[2:-1].split()[0] for line in list_string.split("\n")[1:-1]]
+    return [line[2:-1].split()[0] for line in list_string.split("\n")[1:-2]]
```
Comment (qilinkang) [inline silk/tools/wpan_table_parser.py:450]: Please run other test cases for this change.

File: silk/tests/openthread/ot_test_off_mesh_route_traffic.py (hunk 3 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_off_mesh_route_traffic.py:154]: Ack

File: silk/tools/wpan_table_parser.py (hunk 5 above)
Comment (priyankaCh22) [reply silk/tools/wpan_table_parser.py:450]: Tested with other relevant tests(slaac tests, inform_parent,parent_reset_child_recovery, child_attach_with_multiple_ip_addr) all are PASS.

File: silk/tests/openthread/ot_test_off_mesh_route_traffic.py (hunk 4 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_off_mesh_route_traffic.py]: Ack. This fixed the issue.

File: silk/tests/openthread/ot_test_off_mesh_route_traffic.py (hunk 1 above)
Comment (priyankaCh22) [reply silk/tests/openthread/ot_test_off_mesh_route_traffic.py:228]: Not blocked by issue #68

**Review by qilinkang — APPROVED** [review]
LGTM

**Review by tzm41 — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**